var/spack/repos/builtin/packages/libcatalyst/package.py

```diff
@@ -59,19 +61,33 @@ def setup_run_environment(self, env):
     @on_package_attributes(run_tests=True)
     @run_after("install")
     def build_test(self):
-        testdir = "smoke_test_build"
+        testdir = join_path(self.stage.source_path, "smoke_test_build")
         cmakeExampleDir = join_path(self.stage.source_path, "examples")
         cmake_args = [
             cmakeExampleDir,
             "-DBUILD_SHARED_LIBS=ON",
             self.define("CMAKE_PREFIX_PATH", self.prefix),
         ]
+        adapter0_test_path = join_path(testdir, "adaptor0/adaptor0_test")
+        if sys.platform == "win32":
+            # Default generator on Windows is visual studio, Spack uses
+            # ninja
+            # If visual studio is used the path to the test exe is
+            # adapter0/<CONFIG>/adaptor0_test
+            # due to the nature of VS as a multi-config generator
+            # Use ninja to keep the path/logic the same here
+            # This package is a CMake package so we're already guaranteed to have
+            # ninja in the DAG, so no need to re-depend on it for this purpose
```

Comment:
I suggest the following as a comment:

```
Specify ninja generator for `cmake` call used to generate test artifact
(this differs from the build of `libcatalyst` itself); if unspecified, the
default is to use Visual Studio, which generates a more-complex path
(adapter0/<CONFIG>/adaptor0_test rather than adaptor0/adaptor0_test).
```

* Skips the last part about ninja dep (nothing was wrong with it, but the comment is long and this is a more-generic issue)
* I got confused relating this to e.g. the `generator()` directive, which is not used here